core/string/ustring.cpp

```diff
@@ -161,6 +161,16 @@ bool CharString::operator<(const CharString &p_right) const {
 	return is_str_less(get_data(), p_right.get_data());
 }
 
+bool CharString::operator==(const CharString &p_right) const {
+	if (length() == 0) {
+		return p_right.length() != 0;
```

Comment:
This seems backwards?

Reply:
Warrant adding a test case that would have caught this :)

Reply:
Ah, yeah, brainfart :)